main.go

```diff
@@ -41,9 +43,30 @@ func main() {
 		panic(fmt.Sprintf("Cannot run with a thread number greater than %d", tempThreadLimit))
 	}
 
+	yamlParam := "yaml"
+	yamlValue := dummyYAML
+	yamlUsage := "the path to the YAML configuration file which will supersede all other flags"
+	yaml := flag.String(yamlParam, yamlValue, yamlUsage)
+
 	// parsing the flags has to be done after setting up all the flags
 	flag.Parse()
 
+	var (
+		cfg *config_parser.Config
+		err error
+	)
+	if *yaml != dummyYAML { // if YAML config file was provided
+		cfg, err = config_parser.ParseConfig(*yaml)
+		if err != nil {
+			panic(fmt.Sprintf("the YAML config file was incorrect, err: %v", err))
+		}
+
+		// overriding the GenoMagic core flags if a YAML file is provided
+		*rawSequenceFile = cfg.GenoMagic.InputFilePath
```

Comment:
There's no need to set the variables to the `Config` values as the code already has access to everything.

Reply:
Yes, but the `input` and `output` flags are mandatory and can be the only flags passed in if the YAML configuration file is not used.

Do you suggest I add a check whether or not YAML file is used and create new master accordingly? As follows?

```
if *yaml != dummyYAML {
    mst := master.New(cfg.GenoMagic.inputFilePath, cfg.GenoMagic.outputPath, cfg)
} else {
    mst := master.New(*rawSequenceFile, *out, cfg)
}
```

Reply:
But there's no need for flags. All Geno needs is a YAML file, nothing else.